src/views/components/omnibar.jsx

```diff
@@ -13,16 +15,29 @@ export default class Omnibar extends React.Component {
 
     changeQuery(e) {
         let regex = new RegExp("^[a-zA-Z0-9]*$");
+        let regexOnlyNums = new RegExp("^[0-9]*$");
         if(regex.test(e.target.value)) {
-            if (e.target.value.length === 6) {
+            try {
+                let result = readAddress(e.target.value);
                 this.setState({
                     query: ''
                 });
-                OmnibarController.getStudent(e.target.value);
-            } else {
-                this.setState({
-                    query: e.target.value
-                });
+                if(result.type === 'model') {
+                    ModelController.getModelAndItems(e.target.value);
+                } else if (result.type === 'item') {
+                    OmnibarController.displayItem(e.target.value);
```

Comment:
This is a bit smelly; why is it `ModelController` for showing a model & items, and `OmnibarController` for displaying an item.